docs/static/advanced-pipeline.asciidoc

```diff
@@ -145,60 +134,75 @@ After processing, the sample line has the following JSON representation:
-Elasticsearch cluster. Edit the `first-pipeline.conf` file to add the following text after the `input` section:
+In addition to parsing log data for better searches, filter plugins can derive supplementary information from existing
+data. As an example, the {logstash}plugins-filters-geoip.html[`geoip`] plugin looks up IP addresses, derives geographic
+location information from the addresses, and adds that location information to the logs.
+
+Configure your Logstash instance to use the `geoip` filter plugin by adding the following lines to the `filter` section
+of the `first-pipeline.conf` file:
 
 [source,json]
 --------------------------------------------------------------------------------
-output {
-    elasticsearch {
+    geoip {
+        source => "clientip"
     }
-}
 --------------------------------------------------------------------------------
 
-With this configuration, Logstash uses http protocol to connect to Elasticsearch. The above example assumes Logstash
-and Elasticsearch to be running on the same instance. You can specify a remote Elasticsearch instance using `hosts`
-configuration like `hosts => "es-machine:9092"`.
+The `geoip` plugin configuration requires you to specify the name of the source field that contains the IP address to look up. In this example, the `clientip` field contains the IP address.
 
-[float]
-[[configuring-geoip-plugin]]
-===== Enhancing Your Data with the Geoip Filter Plugin
+Make sure that the `geoip` section is after the `grok` section of the configuration file and that both the `grok` and
+`geoip` sections are nested within the `filter` section like this:
 
-In addition to parsing log data for better searches, filter plugins can derive supplementary information from existing
-data. As an example, the {logstash}plugins-filters-geoip.html[`geoip`] plugin looks up IP addresses, derives geographic
-location information from the addresses, and adds that location information to the logs.
+[source,json]
+--------------------------------------------------------------------------------
+ filter {
+    grok {
+        match => { "message" => "%{COMBINEDAPACHELOG}"}
+    }
+    geoip {
+        source => "clientip"
+    }
+--------------------------------------------------------------------------------
 
-Configure your Logstash instance to use the `geoip` filter plugin by adding the following lines to the `filter` section
-of the `first-pipeline.conf` file:
+
+[float]
+[[indexing-parsed-data-into-elasticsearch]]
+==== Indexing Your Data into Elasticsearch
+
+Now that the web logs are broken down into specific fields, the Logstash pipeline can index the data into an
+Elasticsearch cluster. Edit the `first-pipeline.conf` file and replace the entire `output` section with the following
+text:
 
 [source,json]
 --------------------------------------------------------------------------------
-geoip {
-    source => "clientip"
+output {
+    elasticsearch {
+        hosts => "localhost:9200"
```

Comment:
although it is not necessary, I wonder if we should use `hosts => [ "localhost:9200" ]` to encourage the idea of more nodes..